azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/service/sdk/CallingSDKWrapper.kt

```diff
@@ -252,34 +268,26 @@ internal class CallingSDKWrapper(
         cameraState: CameraState,
         audioState: AudioState,
     ): CompletableFuture<Void> {
-
         val startCallCompletableFuture = CompletableFuture<Void>()
         createCallAgent().thenAccept { agent: CallAgent ->
-            val audioOptions = AudioOptions()
+            val audioOptions = OutgoingAudioOptions()
             audioOptions.isMuted = (audioState.operation != AudioOperationalStatus.ON)
-            val callLocator: JoinMeetingLocator = when (callConfig.callType) {
-                CallType.GROUP_CALL -> GroupCallLocator(callConfig.groupId)
-                CallType.TEAMS_MEETING -> TeamsMeetingLinkLocator(callConfig.meetingLink)
-                /* <ROOMS_SUPPORT:3> */
-                CallType.ROOMS_CALL -> RoomCallLocator(callConfig.roomId)
-                /* </ROOMS_SUPPORT:1> */
-            }
-            var videoOptions: VideoOptions? = null
             // it is possible to have camera state not on, (Example: waiting for local video stream)
             // if camera on is in progress, the waiting will make sure for starting call with right state
             if (camerasCountStateFlow.value != 0 && cameraState.operation != CameraOperationalStatus.OFF) {
                 getLocalVideoStream().whenComplete { videoStream, error ->
+                    val videoOptions = OutgoingVideoOptions()
                     if (error == null) {
                         val localVideoStreams =
                             arrayOf(videoStream.native as NativeLocalVideoStream)
-                        videoOptions = VideoOptions(localVideoStreams)
+                        videoOptions.setOutgoingVideoStreams(localVideoStreams.asList())
                     }
-                    joinCall(agent, audioOptions, videoOptions, callLocator)
+                    connectCall(agent, audioOptions, videoOptions)
```

Comment:
Could we move `connectCall(agent, audioOptions, videoOptions)` out of if-else statement?
if-else should only decide that videoOptions will be.

Reply:
No Pavel, if video options not required, call call can be started immediately, otherwise needs to wait for getLocalVideoStream